### PR TITLE
:sparkles: Add Sonatype Lift as a dependency update tool, doc upgrade

### LINKED ISSUE #2312: Add Lift as a recognized dependency update tool
**Is your feature request related to a problem? Please describe.**
I would like to add Lift (lift.sonatype.com/) as a recognized tool for the Dependency-Update-Tool test to expand the list of popular available tools.

**Describe the solution you'd like**
Add a new case to the Dependency-Update-Tool to pass if a valid configuration file for the Lift tool is found.

**Describe alternatives you've considered**
None in this case.

**Additional context**.

### PR DESCRIPTION
**Pull request by theresa-m**
#### What kind of change does this PR introduce?
This change adds Sonatype Lift as a dependency update tool and updates documentation to include Lift and the previously added PyUp tool.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
dependabot, renovatebot, PyUp are recognized dependency upgrade tools.

#### What is the new behavior (if this is a feature change)?**
Configuration files for Sonatype Lift are detected as an option for the dependency update tool check.

- [x] Tests for the changes have been added (for bug fixes/features)
_I've added Lift configuration file names to an existing test although it doesn't appear to actually run. Please advise if this test was intended to be removed._

#### Which issue(s) this PR fixes
#2312 

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
Projects using Sonatype Lift will pass the Dependency Update Tool check if a .lift/config.toml and .lift.toml exists.
```

### CODE REVIEW

**Comment by azeemshaikh38** [comment]
@theresa-m looking at https://help.sonatype.com/lift, Sonatype Lift seems to be a code analysis tool and not really something that helps with automatic dependency updates?

**Comment by theresa-m** [comment]
hey @azeemshaikh38 Lift notifies projects of new and existing vulnerabilities in dependencies which to me seems in the spirit of keeping vulnerable dependencies up to date.

**Review by azeemshaikh38 — APPROVED** [review]

File: docs/checks/internal/checks.yaml [hunk 1]
```diff
@@ -57,9 +57,12 @@ checks:
       Risk: `High` (possibly vulnerable to attacks on known flaws)
 
       This check tries to determine if the project uses a dependency update tool,
-      specifically [dependabot](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates) or
-      [renovatebot](https://docs.renovatebot.com/configuration-options/). Out-of-date
-      dependencies make a project vulnerable to known flaws and prone to attacks.
+      specifically one of:
```
Comment (azeemshaikh38) [inline docs/checks/internal/checks.yaml:60]: Please run `make generate-docs`.

**Review by azeemshaikh38 — APPROVED** [review]

File: docs/checks/internal/checks.yaml (hunk 1 above)
Comment (theresa-m) [reply docs/checks/internal/checks.yaml:60]: Done. Sorry for the delay.

**Pull request merged**